### PR TITLE
Fix two different duplicate contributor messages. Fixes #1082

### LINKED ISSUE #1082: Duplicate contributor error message
When trying to add the same contributor to a course twice, an error message is shown. This message differs based on whether a questionnaire was assigned or not. The error message should in both cases be the one currently shown when no questionnaire was selected.

**Current messages:**
Questionnaire selected: `Please correct the duplicate data for contributor. Please correct the duplicate values below.`
No questionnaire selected: `Duplicate contributor found. Each contributor should only be used once.`

### PR DESCRIPTION
**Pull request by FSadrieh**
Fixes #1082

### CODE REVIEW

File: evap/staff/forms.py [hunk 1]
```diff
@@ -574,6 +574,8 @@ def handle_deleted_and_added_contributions(self):
                 # we modified the form, so we have to force re-validation
                 form_with_errors.full_clean()
 
+
+
```
Comment (janno42) [inline evap/staff/forms.py:578]: please remove these extra lines

**Review by janno42 — CHANGES_REQUESTED** [review]
Nice! Just a very small change request.

**Comment by FSadrieh** [comment]
I hope that fixes it

File: evap/staff/forms.py (hunk 1 above)
Comment (FSadrieh) [reply evap/staff/forms.py]: I hope I fixed it

File: evap/staff/forms.py [hunk 2]
```diff
@@ -632,6 +629,7 @@ def clean(self):
                 raise forms.ValidationError(_('Duplicate contributor found. Each contributor should only be used once.'))
```
Comment (janno42) [inline evap/staff/forms.py:629]: you could add the name of the duplicated contributor to the message like this:
```
raise forms.ValidationError(_('Duplicate contributor ({}) found. Each contributor should only be used once.').format(contributor.full_name))
```

**Review by janno42 — APPROVED** [review]
Very nice first PR!
I noticed one additional thing you could do to make the error message even better than before if you want :)

File: evap/staff/tests/test_forms.py [hunk 3]
```diff
@@ -257,13 +257,22 @@ def test_contribution_form_set(self):
...
-        data['contributions-1-questionnaires'] = questionnaire.pk
         data['contributions-1-order'] = 1
         data['contributions-1-textanswer_visibility'] = Contribution.TextAnswerVisibility.GENERAL_TEXTANSWERS
-        self.assertFalse(ContributionFormset(instance=evaluation, form_kwargs={'evaluation': evaluation}, data=data).is_valid())
+        data['contributions-1-role'] = Contribution.Role.EDITOR
+        formset = ContributionFormset(instance=evaluation, form_kwargs={'evaluation': evaluation}, data=data)
+        self.assertFalse(formset.is_valid())
+        # regression for https://github.com/e-valuation/EvaP/issues/1082
+        # assert same error message with and without questionnaire
+        self.assertEqual(formset.non_form_errors(), ['Duplicate contributor(' + contributions-1-contributor ')found. Each contributor should only be used once.'])
```
Comment (janno42) [inline evap/staff/tests/test_forms.py:267]: Similar to how you created the message in the forms.py, you should use `contributor.full_name` as the string inserted into the error message. Pay attention: You forgot the second `+` and the spaces around the parentheses.

**Review by janno42 — APPROVED** [review]
thumbs up!

**Pull request merged**